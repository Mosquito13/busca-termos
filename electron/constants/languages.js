module.exports = {
  USA: 'en-us',
  BRAZIL: 'pt-br',
  SPAIN: 'es',
  PORTUGAL: 'pt',
  GERMANY: 'de',
  FRANCE: 'fr',
  ROMANIA: 'ro',
  ITALIA: 'it',
  SLOVAKIA: 'sk',
  TURKEY: 'tr',
  RUSSIA: 'ru',
  POLAND: 'pl',
  CATALONIA: 'ca',
  DENMARK: 'da',
  FINLAND: 'fi',
  CZECH: 'cs'
};
