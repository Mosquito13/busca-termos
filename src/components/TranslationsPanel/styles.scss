@import '../../sass/styles.scss';

.translations-panel {
  flex: 1;
  height: 100%;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 10px rgba($color-black, 0.3);
  border-radius: $radius;
  overflow: hidden;
  @include themify() {
    background-color: themed('color-base-lighter');
    border: 1px solid themed('color-border');
  }

  &__header {
    flex: 0 0 3rem;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: content-box;
    @include themify() {
      border-bottom: 1px solid themed('color-border');
      background-color: themed('color-base-darker');
    }
  }

  &__fields {
    flex: 1;
    display: flex;
    flex-direction: column;
    padding: 0.5rem;
    overflow-y: auto;

    @include small-height-app() {
      padding: 0;
    }
  }

  &__field-row {
    display: flex;

    &:not(:last-child) {
      @include small-height-app() {
        @include themify() {
          border-bottom: 1px solid themed('color-border');
        }
      }
    }
  }

  &__field-row > &__field:nth-child(1),
  &__field-row > &__field:nth-child(2) {
    @include small-height-app() {
      @include themify() {
        border-right: 1px solid themed('color-border');
      }
    }
  }

  &__field {
    margin: 0.5rem;
    overflow: hidden;
    border-width: 1px;
    border-style: solid;
    border-radius: $radius;
    @include themify() {
      border-color: themed('color-border');
      background-color: themed('color-base');
    }

    @include small-height-app() {
      margin: 0;
      border-radius: 0;
      border: none;
    }

    &--columns-1 {
      flex: 1;
    }

    &--columns-2 {
      flex: 0 0 calc(50% - 1rem);
      @include small-height-app() {
        flex: 0 0 50%;
      }
    }

    &--columns-3 {
      flex: 0 0 calc(33.333% - 1rem);
      @include small-height-app() {
        flex: 0 0 33.333%;
      }
    }
  }

  &__empty-state {
    margin: 1rem;
    text-align: center;
  }
}
