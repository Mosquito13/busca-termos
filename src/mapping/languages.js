import { ReactComponent as BrazilIcon } from '../assets/svg/brazil.svg';
import { ReactComponent as CataloniaIcon } from '../assets/svg/catalonia.svg';
import { ReactComponent as CzechIcon } from '../assets/svg/czech.svg';
import { ReactComponent as DenmarkIcon } from '../assets/svg/denmark.svg';
import { ReactComponent as FinlandIcon } from '../assets/svg/finland.svg';
import { ReactComponent as FranceIcon } from '../assets/svg/france.svg';
import { ReactComponent as GermanyIcon } from '../assets/svg/germany.svg';
import { ReactComponent as ItalyIcon } from '../assets/svg/italy.svg';
import { ReactComponent as PolandIcon } from '../assets/svg/poland.svg';
import { ReactComponent as PortugalIcon } from '../assets/svg/portugal.svg';
import { ReactComponent as RomaniaIcon } from '../assets/svg/romania.svg';
import { ReactComponent as RussiaIcon } from '../assets/svg/russia.svg';
import { ReactComponent as SlovakiaIcon } from '../assets/svg/slovakia.svg';
import { ReactComponent as SpainIcon } from '../assets/svg/spain.svg';
import { ReactComponent as TurkeyIcon } from '../assets/svg/turkey.svg';
import { ReactComponent as UsaIcon } from '../assets/svg/usa.svg';

const languageMapping = {
  BRAZIL: {
    id: 'pt-br',
    title: 'Português (Brasil)',
    getIcon: () => <BrazilIcon />
  },
  CATALONIA: {
    id: 'ca',
    title: 'Catalão',
    getIcon: () => <CataloniaIcon />
  },
  CZECH: {
    id: 'cs',
    title: 'Tcheco',
    getIcon: () => <CzechIcon />
  },
  DENMARK: {
    id: 'da',
    title: 'Dinamarquês',
    getIcon: () => <DenmarkIcon />
  },
  FINLAND: {
    id: 'fi',
    title: 'Finlandês',
    getIcon: () => <FinlandIcon />
  },
  FRANCE: {
    id: 'fr',
    title: 'Francês',
    getIcon: () => <FranceIcon />
  },
  GERMANY: {
    id: 'de',
    title: 'Alemão',
    getIcon: () => <GermanyIcon />
  },
  ITALY: {
    id: 'it',
    title: 'Italiano',
    getIcon: () => <ItalyIcon />
  },
  POLAND: {
    id: 'pl',
    title: 'Polonês',
    getIcon: () => <PolandIcon />
  },
  PORTUGAL: {
    id: 'pt',
    title: 'Português (Portugal)',
    getIcon: () => <PortugalIcon />
  },
  ROMANIA: {
    id: 'ro',
    title: 'Romeno',
    getIcon: () => <RomaniaIcon />
  },
  RUSSIA: {
    id: 'ru',
    title: 'Russo',
    getIcon: () => <RussiaIcon />
  },
  SLOVAKIA: {
    id: 'sk',
    title: 'Eslovaco',
    getIcon: () => <SlovakiaIcon />
  },
  SPAIN: {
    id: 'es',
    title: 'Espanhol',
    getIcon: () => <SpainIcon />
  },
  TURKEY: {
    id: 'tr',
    title: 'Turco',
    getIcon: () => <TurkeyIcon />
  },
  USA: {
    id: 'en-us',
    title: 'Inglês (EUA)',
    getIcon: () => <UsaIcon />
  }
};

export { languageMapping };
