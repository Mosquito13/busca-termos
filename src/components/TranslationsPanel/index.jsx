import { useSelector } from 'react-redux';
import orderBy from 'lodash/orderBy';

import { languageMapping } from '../../mapping/languages';
import TranslationField from '../TranslationField';
import Heading4 from '../common/Heading4';

import settingsSelectors from '../../selectors/settings';

import './styles.scss';

const TranslationsPanel = () => {
  const mainLanguage = useSelector(settingsSelectors.getMainLanguage);
  const translation = useSelector(settingsSelectors.getTranslation);

  const fields = [];

  orderBy(languageMapping, ['title']).forEach(({ id, title, getIcon }) => {
    if (id !== mainLanguage && translation[id]) {
      fields.push(
        <div key={id} className="translations-panel__field">
          <TranslationField
            id={id}
            title={title}
            FlagIcon={getIcon()}
          />
        </div>
      );
    }
  });

  return (
    <div className="translations-panel">
      <div className="translations-panel__header">
        <Heading4>{'Traduções'}</Heading4>
      </div>
      <div className="translations-panel__fields">
        {fields}
      </div>
    </div>
  );
};

export default TranslationsPanel;
