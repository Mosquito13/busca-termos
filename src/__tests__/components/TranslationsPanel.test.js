import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';

import TranslationsPanel from '../../components/TranslationsPanel';
import { languageMapping } from '../../mapping/languages';

const stateMock = {
  Core: {
    selectedItemId: 321
  },
  Settings: {
    mainLanguage: languageMapping.USA.id,
    translation: {
      [languageMapping.USA.id]: true,
      [languageMapping.SPAIN.id]: true,
      [languageMapping.BRAZIL.id]: true,
      [languageMapping.FRANCE.id]: true,
      [languageMapping.FINLAND.id]: true
    }
  }
};

const prepareReactReduxMocks = () => {
  jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => jest.fn());
  jest
    .spyOn(reactRedux, 'useSelector')
    .mockImplementation((selector) => selector(stateMock));
};

describe(`${TranslationsPanel.name}`, () => {
  describe('tests', () => {
    let wrapper;

    beforeAll(() => {
      prepareReactReduxMocks();

      wrapper = shallow(<TranslationsPanel />);
    });

    it('should render with selected fields', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.translations-panel__field-row')).toHaveLength(2);
      expect(wrapper.find('.translations-panel__field')).toHaveLength(4);
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      prepareReactReduxMocks();

      const tree = renderer.create(<TranslationsPanel />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
