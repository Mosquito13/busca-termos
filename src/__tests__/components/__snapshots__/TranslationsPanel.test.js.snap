// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationsPanel snapshots should render empty state when does not have fields enabled 1`] = `
<div
  className="translations-panel"
>
  <div
    className="translations-panel__header"
  >
    Traduções
  </div>
  <div
    className="translations-panel__empty-state"
  >
    <p
      className="paragraph"
    >
      Você pode habilitar traduções através da página de preferências.
    </p>
  </div>
</div>
`;

exports[`TranslationsPanel snapshots should render with 1 column 1`] = `
<div
  className="translations-panel"
>
  <div
    className="translations-panel__header"
  >
    Traduções
  </div>
  <div
    className="translations-panel__fields"
  >
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-1"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Espanhol"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  spain.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-1"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Finlandês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  finland.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-1"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Francês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  france.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-1"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Português (Brasil)"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  brazil.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TranslationsPanel snapshots should render with 2 columns 1`] = `
<div
  className="translations-panel"
>
  <div
    className="translations-panel__header"
  >
    Traduções
  </div>
  <div
    className="translations-panel__fields"
  >
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-2"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Espanhol"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  spain.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="translations-panel__field translations-panel__field--columns-2"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Finlandês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  finland.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-2"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Francês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  france.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="translations-panel__field translations-panel__field--columns-2"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Português (Brasil)"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  brazil.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TranslationsPanel snapshots should render with 3 columns 1`] = `
<div
  className="translations-panel"
>
  <div
    className="translations-panel__header"
  >
    Traduções
  </div>
  <div
    className="translations-panel__fields"
  >
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-3"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Espanhol"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  spain.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="translations-panel__field translations-panel__field--columns-3"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Finlandês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  finland.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="translations-panel__field translations-panel__field--columns-3"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Francês"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  france.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="translations-panel__field-row"
    >
      <div
        className="translations-panel__field translations-panel__field--columns-3"
      >
        <div
          className="translation-field"
        >
          <div
            className="translation-field__flag"
          >
            <div
              className="translation-field__icon"
              title="Português (Brasil)"
            >
              <div
                className="icon icon--small"
              >
                <svg>
                  brazil.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="input-text"
          >
            <input
              className="input-text__input input-text__input--disabled input-text__input--borderless"
              disabled={true}
              onChange={[Function]}
              placeholder="Selecione um item na lista"
              spellCheck={false}
              type="text"
              value=""
            />
            <div
              className="input-text__copy-button"
            >
              <div
                className="copy-text-button"
                onClick={[Function]}
                title="Copiar"
              >
                <div
                  className="icon icon--small"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="13"
                      rx="2"
                      ry="2"
                      width="13"
                      x="9"
                      y="9"
                    />
                    <path
                      d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
